server/Arcadia.Assistant.Calendar.Abstractions/EventBus/CalendarEventApprovalsChanged.cs

```diff
@@ -1,17 +1,38 @@
 ﻿namespace Arcadia.Assistant.Calendar.Abstractions.EventBus
 {
+    using System;
     using System.Collections.Generic;
 
     public class CalendarEventApprovalsChanged
     {
-        public CalendarEventApprovalsChanged(CalendarEvent @event, IEnumerable<string> approvals)
+        public CalendarEventApprovalsChanged(
+            CalendarEvent @event,
+            string lastApprovedBy,
+            DateTimeOffset lastApprovalTimestamp,
+            IEnumerable<string> approvals)
         {
             this.Event = @event;
+            this.LastApproval = new Approval(lastApprovedBy, lastApprovalTimestamp);
             this.Approvals = approvals;
         }
 
         public CalendarEvent Event { get; }
 
+        public Approval LastApproval { get; set; }
+
         public IEnumerable<string> Approvals { get; }
```

Comment:
why don't you have IEnumerable<Approval> Approvals instead. Then you can always find LastApproval by timestamp.

Reply:
Done.